src/modules/profile/infos/name.tsx

```diff
@@ -41,24 +47,39 @@ export const ProfileInfosName = observer(() => {
 
       <TextInput
         label="First Name"
-        value={profileFlowStore.firstName}
-        onChangeText={(firstName) => (profileFlowStore.firstName = firstName)}
+        value={firstName}
+        onChangeText={firstName => SetFirstName(firstName)}
         autoCorrect={false}
-        placeholder="John"
         // placeholderTextColor={colors.blue}
       />
 
       <TextInput
         label="Last Name"
-        value={profileFlowStore.lastName}
-        onChangeText={(lastName) => (profileFlowStore.lastName = lastName)}
+        value={lastName}
+        onChangeText={lastName => SetLastName(lastName)}
         autoCorrect={false}
         placeholder="Doe"
         // placeholderTextColor={colors.blue}
       />
 
       <View style={styles.buttonContainer}>
-        <Button onPress={navigation.goBack}>Done</Button>
+        <Button
+          onPress={async () => {
+            if (userStore.profile._id) {              
+              await userStore.patchProfile(userStore.profile._id, {
```

Comment:
I don't see why we await we the end of the result. Let's do it in "optimistic update" mode